counterpartylib/lib/util.py

```diff
@@ -198,6 +202,79 @@ def get_asset_name (db, asset_id, block_index):
     elif not assets:
         return 0    # Strange, I know…
 
+# checks and validates subassets
+#   throws exceptions for assset or subasset names with invalid syntax
+def parse_subasset_from_asset_name(asset):
+    subasset_parent = None
+    subasset_child = None
+    subasset = None
+    chunks = asset.split('.', 1)
+    if (len(chunks) == 2):
+        subasset_parent = chunks[0]
+        subasset_child = chunks[1]
+        subasset = asset
+
+        # validate parent asset
+        validate_subasset_parent_name(subasset_parent)
+
+        # validate child asset
+        validate_subasset_name(subasset, subasset_child)
+
+    return (subasset_parent, subasset)
+
+# takes an asset description in 
+#   if it has a suffix like ;;lPARENT.child, then parse the subasset information
+#   throws exceptions for invalid asset names
+def parse_subasset_from_description(description):
+    subasset_parent = None
+    subasset_child = None
+    subasset = None
+    chunks = description.rsplit(';;l', 1)
+    if len(chunks) == 2:
+        split_description, asset = chunks
+        subasset_parent, subasset = parse_subasset_from_asset_name(asset)
+        if subasset is not None:
+            # if we parsed a subasset, strip ;;l{subasset} from the description
+            description = split_description
+
+    return (subasset_parent, subasset, description)
+
+# throws exceptions for invalid subasset names
+def validate_subasset_name(subasset, subasset_child):
```

Comment:
`subasset_longname` = PIZZA.DOMINOS
`subasset_parent` = PIZZA
`subasset_child` = DOMINOS

`subasset_child` is only used in 2 validation functions.